crates/quickjs-wasm-rs/Cargo.toml

```diff
@@ -17,6 +17,7 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = { version = "1.0", optional = true }
 serde-transcode = { version = "1.1", optional = true }
 once_cell = "1.16"
+thiserror = "1.0"
```

Comment:
I'd like to avoid this dependency if possible. I think it's still valuable to be conscious about the generated code size which we still want to keep as small as possible in all cases (dynamic vs static modules).

Since `anyhow` works with any type that implements `std::error::Error` is there any reason, on why we can't instead manually implement `Error` for `JSError`?

Reply:
That's fair. I originally wrote an implementation that manually implemented `Error` but it was messy. I added `thiserror` which cleaned it up, then cleaned up some other aspects of how it was implemented, and didn't revisit removing the dependency with the cleaned up code. The version without the dependency I just pushed up looks nicer than with `thiserror` so that's good.